Composer/packages/extensions/obiformeditor/src/Form/fields/PromptField/ChoiceInput/Choices.tsx

```diff
@@ -3,239 +3,74 @@
 
 /** @jsx jsx */
 import { jsx } from '@emotion/core';
-import React, { useState } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { Dropdown, ResponsiveMode } from 'office-ui-fabric-react/lib/Dropdown';
 import { JSONSchema6 } from 'json-schema';
-import formatMessage from 'format-message';
-import { TextField } from 'office-ui-fabric-react/lib/TextField';
-import { IconButton } from 'office-ui-fabric-react/lib/Button';
-import { IContextualMenuItem } from 'office-ui-fabric-react/lib/ContextualMenu';
-import { NeutralColors, FontSizes, SharedColors } from '@uifabric/fluent-theme';
 import { IChoice } from '@bfc/shared';
+import formatMessage from 'format-message';
 
-import {
-  field,
-  choiceField,
-  choiceItem,
-  choiceItemContainer,
-  choiceItemLabel,
-  choiceItemValue,
-  choiceItemValueLabel,
-} from '../styles';
-import { swap, remove } from '../../../utils';
 import { WidgetLabel } from '../../../widgets/WidgetLabel';
-import { EditableField } from '../../EditableField';
 
-interface ChoiceItemProps {
-  index: number;
-  choice: IChoice;
-  hasMoveUp: boolean;
-  hasMoveDown: boolean;
-  onReorder: (a: number, b: number) => void;
-  onDelete: (idx: number) => void;
-  onEdit: (idx: number, value?: IChoice) => void;
-}
+import { StaticChoices } from './StaticChoices';
+import { DynamicChoices } from './DynamicChoices';
 
 interface ChoicesProps {
   id: string;
   schema: JSONSchema6;
-  formData?: IChoice[];
+  formData?: IChoice[] | string;
```

Comment:
This maybe should be defined in the shared types:

```ts
type IChoice = ObjectChoice[] | string;
```

Choose a better name for `ObjectChoice`, but you get the idea.